src/Nethermind/Nethermind.Taiko/Rpc/TaikoEngineRpcModule.cs

```diff
@@ -172,7 +172,6 @@ void CommitAndDisposeBatch(Batch batch)
         }
 
         BlockExecutionContext blkCtx = new(blockHeader, _specProvider.GetSpec(blockHeader));
-        worldState.SetBaseBlock(blockHeader);
```

Comment:
In line 151, the `blockHeader` state root is set to `head.StateRoot` which the worldstate is already set by `txProcessorSource` in line 137.